From 242f9ae241b1917cab412c3decc6bfc3c18f6a55 Mon Sep 17 00:00:00 2001
From: Jonas Gorski <jonas.gorski@bisdn.de>
Date: Fri, 26 Aug 2022 14:16:09 +0200
Subject: [PATCH 7/8] accton-as4610: fix FAN driver id table

This is a platform driver, not an I2C driver, so fix and add the id
table to the driver.

Upstream-Status: Backport [https://github.com/opencomputeproject/OpenNetworkLinux/commit/58af58fdede50ec97b206eb22b123432f2c24092]
Signed-off-by: Jonas Gorski <jonas.gorski@bisdn.de>
---
 .../armxx/arm-accton-as4610/modules/accton_as4610_fan.c      | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/packages/platforms/accton/armxx/arm-accton-as4610/modules/accton_as4610_fan.c b/packages/platforms/accton/armxx/arm-accton-as4610/modules/accton_as4610_fan.c
index 2e14798a9721..ff75c0f9d464 100644
--- a/packages/platforms/accton/armxx/arm-accton-as4610/modules/accton_as4610_fan.c
+++ b/packages/platforms/accton/armxx/arm-accton-as4610/modules/accton_as4610_fan.c
@@ -299,15 +299,16 @@ static int as4610_fan_remove(struct platform_device *pdev)
 	return 0;
 }
 
-static const struct i2c_device_id as4610_fan_id[] = {
+static const struct platform_device_id as4610_fan_id[] = {
 	{ "as4610_fan", 0 },
 	{}
 };
-MODULE_DEVICE_TABLE(i2c, as4610_fan_id);
+MODULE_DEVICE_TABLE(platform, as4610_fan_id);
 
 static struct platform_driver as4610_fan_driver = {
 	.probe		= as4610_fan_probe,
 	.remove		= as4610_fan_remove,
+	.id_table	= as4610_fan_id,
 	.driver		= {
 		.name	= DRVNAME,
 		.owner	= THIS_MODULE,
-- 
2.42.0

